src/auto/dbus_connection.rs

```diff
@@ -0,0 +1,1264 @@
+        unsafe {
+            from_glib(gio_sys::g_dbus_connection_get_exit_on_close(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[cfg(any(feature = "v2_60", feature = "dox"))]
+    pub fn get_flags(&self) -> DBusConnectionFlags {
+        unsafe { from_glib(gio_sys::g_dbus_connection_get_flags(self.to_glib_none().0)) }
+    }
+
+    pub fn get_guid(&self) -> Option<GString> {
+        unsafe { from_glib_none(gio_sys::g_dbus_connection_get_guid(self.to_glib_none().0)) }
+    }
+
+    pub fn get_last_serial(&self) -> u32 {
+        unsafe { gio_sys::g_dbus_connection_get_last_serial(self.to_glib_none().0) }
+    }
+
+    pub fn get_peer_credentials(&self) -> Option<Credentials> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_connection_get_peer_credentials(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_stream(&self) -> Option<IOStream> {
+        unsafe { from_glib_none(gio_sys::g_dbus_connection_get_stream(self.to_glib_none().0)) }
+    }
+
+    pub fn get_unique_name(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_connection_get_unique_name(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn is_closed(&self) -> bool {
+        unsafe { from_glib(gio_sys::g_dbus_connection_is_closed(self.to_glib_none().0)) }
+    }
+
+    //pub fn register_object(&self, object_path: &str, interface_info: &DBusInterfaceInfo, vtable: /*Ignored*/Option<&DBusInterfaceVTable>, user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> Result<(), glib::Error> {
+    //    unsafe { TODO: call gio_sys:g_dbus_connection_register_object() }
+    //}
+
+    #[cfg(any(feature = "v2_46", feature = "dox"))]
+    pub fn register_object_with_closures(
+        &self,
+        object_path: &str,
+        interface_info: &DBusInterfaceInfo,
+        method_call_closure: Option<&glib::Closure>,
+        get_property_closure: Option<&glib::Closure>,
+        set_property_closure: Option<&glib::Closure>,
+    ) -> Result<(), glib::Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let _ = gio_sys::g_dbus_connection_register_object_with_closures(
+                self.to_glib_none().0,
+                object_path.to_glib_none().0,
+                interface_info.to_glib_none().0,
+                method_call_closure.to_glib_none().0,
+                get_property_closure.to_glib_none().0,
+                set_property_closure.to_glib_none().0,
+                &mut error,
+            );
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    //pub fn register_subtree(&self, object_path: &str, vtable: /*Ignored*/&DBusSubtreeVTable, flags: /*Ignored*/DBusSubtreeFlags, user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> Result<(), glib::Error> {
```

Comment:
Curious that there is no closures variant of this one... https://gitlab.gnome.org/GNOME/glib/-/issues/2122